rct229/ruletest_engine/ruletest_jsons/scripts/excel_generation_utilities.py

```diff
@@ -0,0 +1,225 @@
+                                                      "Rule_Unit_Test": ["23-1-a", "23-1-b", ...]
+                                                       etc.
+
+    """
+
+    ruletest_directory = f"../{ruleset_standard}"
+    ruletest_url = f"https://github.com/pnnl/ruleset-checking-tool/tree/develop/rct229/ruletest_engine/ruletest_jsons/{ruleset_standard}"
+
+    # Dictionary mapping sections to a list of rule test JSONs relevant to that section
+    # Format ruletest_dict[SECTION_NAME] = ['ruletest1.json', 'ruletest2.json', ...]
+    ruletest_dict = {}
+
+    # Use os.scandir to get directory entries
+    with os.scandir(ruletest_directory) as entries:
+        for entry in entries:
+            if entry.is_dir() and entry.name.startswith("section"):
+                # Section is last subdirectory
+                subdirectory = entry.path
+                section = subdirectory.split("\\")[-1]
+
+                # Initialize the dictionary as a list then populate it with rule test JSONs
+                ruletest_dict[section] = []
+                ruletest_dict[section].extend(
+                    glob.glob(os.path.join(subdirectory, "rule*.json"))
+                )
+
+    # Reorder sections to be in numerical order (i.e., avoid section1, section11, section12, section5, section6)
+    sections_list = list(ruletest_dict.keys())
+    sorted_sections = sorted(sections_list, key=lambda x: int(re.sub(r"\D", "", x)))
+    ruletest_dict = {key: ruletest_dict[key] for key in sorted_sections}
+
+    # Initialize dict to hold Rules information.
+    # Keys include: Rule, Rule_Unit_Test, Test_Description, Expected_Rule_Outcome, Rule_Unit_Test_JSON
+    master_rule_data_dict = {}
+    rule_data_columns = [
+        "Rule",
+        "Rule_Unit_Test",
+        "Test_Description",
+        "Expected_Rule_Outcome",
+        "Rule_Unit_Test_JSON",
+    ]
+
+    # Aggregate JSON information into master_rule_data_dict
+    # Format: master_rule_data_dict[SECTION] = { rule_data_columns[i]: [x1, x2, x3]}
+    #     EX: master_rule_data_dict["section23"] = {"Rule": ["23-1", "23-1", ...],
+    #                                               "Rule_Unit_Test": ["23-1-a", "23-1-b", ...]
+    #                                                etc.
+    for section_key in ruletest_dict:
+
+        # Initialize each column as a list in rule_data_dict
+        for rule_column in rule_data_columns:
+            if section_key not in master_rule_data_dict:
+                master_rule_data_dict[section_key] = {}
+            master_rule_data_dict[section_key][rule_column] = []
+
+        # Read JSON files and populate master_rule_data_dict
+        for ruletest_json in ruletest_dict[section_key]:
+            with open(ruletest_json, "r") as json_file:
+                ruletest_json_dict = json.load(json_file)
+
+                # Append this test JSONs case information
+                for test_case in ruletest_json_dict:
```

Comment:
Hi Yun Joon,

I suppose it would as you dont need to load the JSON file anymore as it's been loaded into ruletest_json_dict. That's a good catch. I'll unindent this 1-level for clarity.

Reply:
Thank you for checking Juan!